Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/unstructured.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
+)
+
+// UnstructuredJSONScheme is capable of converting JSON data into the Unstructured
+// type, which can be used for generic access to objects without a predefined scheme.
+var UnstructuredJSONScheme ObjectDecoder = unstructuredJSONScheme{}
+
+type unstructuredJSONScheme struct{}
+
+// Recognizes returns true for any version or kind that is specified (internal
+// versions are specifically excluded).
+func (unstructuredJSONScheme) Recognizes(version, kind string) bool {
+	return len(version) > 0 && len(kind) > 0
+}
+
+func (s unstructuredJSONScheme) Decode(data []byte) (Object, error) {
+	unstruct := &Unstructured{}
+	if err := s.DecodeInto(data, unstruct); err != nil {
+		return nil, err
+	}
+	return unstruct, nil
+}
+
+func (unstructuredJSONScheme) DecodeInto(data []byte, obj Object) error {
+	unstruct, ok := obj.(*Unstructured)
+	if !ok {
+		return fmt.Errorf("the unstructured JSON scheme does not recognize %v", reflect.TypeOf(obj))
+	}
+
+	m := make(map[string]interface{})
```

Comment:
----- Original Message -----
> > +
> > +func (s unstructuredJSONScheme) Decode(data []byte) (Object, error) {
> > +	unstruct := &Unstructured{}
> > +	if err := s.DecodeInto(data, unstruct); err != nil {
> > +		return nil, err
> > +	}
> > +	return unstruct, nil
> > +}
> > +
> > +func (unstructuredJSONScheme) DecodeInto(data []byte, obj Object) error {
> > +	unstruct, ok := obj.(*Unstructured)
> > +	if !ok {
> > +		return fmt.Errorf("the unstructured JSON scheme does not recognize %v",
> > reflect.TypeOf(obj))
> > +	}
> > +
> > +	m := make(map[string]interface{})
> 
> Any way we can have this recurse through the values so that an `Unstructured`
> could contain a `Pod` and we'd recognize it?  I don't have a particular
> reason, it just seems like it would be nice to have and we ought to be able
> to recognize a map value that is a `map[string]interface{}` that contains
> both a `kind` and `version` key.

Not sure what you're asking - if you call DecodeInto on UnstructuredJSONScheme you're asking of an unstructured object, and typemeta would be filled out for the pod.  It's just we don't convert it to internal.